lightbluetent/api.py

```diff
@@ -9,26 +10,52 @@
 # Usage is something like this (error handling omitted):
 #
 #   meeting = AttendeeMeeting("meeting1234", "password1234")
-#   success, url = meeting.join("John Smith")
+#   url = meeting.get_join_url("John Smith")
 #   redirect to url
 
 class AttendeeMeeting:
 
-    def __init__(self, meeting_id, attendee_pw):
+    def __init__(self, meeting_id, attendee_pw, logo):
         self.meeting_id = meeting_id
         self.attendee_pw = attendee_pw
         self.bbb = BBB()
 
+        # self.logo is None if the default logo is used
+        if logo != current_app.config["DEFAULT_BBB_LOGO"]:
+            self.logo = logo
+        else:
+            self.logo = None
+
 
-    # Join the meeting with a given full name.
-    # Returns (True, "[join_url]") on success, (False, "[error_msg]") on failure.
-    def join(self, full_name):
+    # Builds and returns the join URL for the meeting. Does not check that the
+    # meeting is running.
+    def get_join_url(self, full_name):
         params = {}
         params["fullName"] = full_name
         params["meetingID"] = self.meeting_id
         params["password"] = self.attendee_pw
+        params["redirect"] = "true"
+
+        if self.logo is not None:
+
+            logo_path = os.path.join(current_app.config["IMAGES_DIR_FROM_STATIC"], self.logo)
+            params["logo"] = url_for("static", filename=logo_path, _external=True)
+
+            params["userdata-bbb_display_branding_area"] = "true"
+
+            # Custom styling to make the bbb_logo look better
+            params["userdata-bbb_custom_style"] = ".branding--Z1T4eH0>img{display:block;margin-right:auto;margin-left:auto;}.separator--Z3YSEe{margin-top:0;}.branding--Z1T4eH0 {padding:var(--sm-padding-x);}"
 
-        return self.bbb.join(params)
+        return self.bbb.build_url("join", params)
+
+
+    # Returns True if meeting is running,
+    # False if meeting is not running or on error.
+    def is_running(self):
```

Comment:
Given that BBB contains all the other implementations of the API, I suggest we put this there too. This way we don't have to two duplicate methods in Attendee and Moderator.

Reply:
And make it a static method too, with just one param, the bbb_id.